Update gops to v0.3.18 and build it statically linked

tklauser opened the pull request:
See commit messages for details.

tklauser committed:
images/runtime: build gops without cgo enabled

Some dependencies of gops use cgo on platform other than Linux (e.g.
macOS and OpenBSD) and thus by default the binary is linked dynamically.
Set CGO_ENABLED=0 to explicitly disable cgo and build a statically
linked binary.

Reported-by: André Martins <andre@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
images, vendor: update gops to 0.3.18

Release notes: https://github.com/google/gops/releases/tag/v0.3.18
Changes: https://github.com/google/gops/compare/v0.3.17...v0.3.18

Signed-off-by: Tobias Klauser <tobias@cilium.io>

pchaigno commented on images/cilium-test/Dockerfile:
```diff
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
-ARG CILIUM_BUILDER_IMAGE=quay.io/cilium/cilium-builder:3203db544891bf8360d97085ecce89282f04c78d@sha256:c648e3e2f9f99b7e53922691cfea7515ebeb7aad273f9738f00103494dabcc32
+ARG CILIUM_BUILDER_IMAGE=quay.io/cilium/cilium-builder:5883692e7dfb1d7b96f31c710d4ef31e6e812654
```
Why is the SHA256 hash dropped here and in `test/k8sT/manifests/test-verifier.yaml`?

tklauser replied:
I'm still in the process of building updated images (see https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images). They will be added again in the final iteration.

tklauser replied:
All images built now, the hashes should be updated as well.

tklauser committed:
images/scripts: update cilium-builder image in tail calls hooks test manifest

The CI tests added in commit 37f6192c9e77 ("test: add CI test for tail
calls hooks for custom programs") use the cilium-builder image. Make
sure the image is updated by the update-cilium-builder-image.sh script
when creating new runtime/builder images.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
images: update cilium-{runtime,builder}

Signed-off-by: Tobias Klauser <tobias@cilium.io>
